src/vehicle.cpp

```diff
@@ -2513,18 +2514,36 @@ void vehicle::print_fuel_indicator (void *w, int y, int x, bool fullsize, bool v
 
 void vehicle::coord_translate (int reldx, int reldy, int &dx, int &dy)
 {
-    tileray tdir (face.dir());
-    tdir.advance (reldx);
-    dx = tdir.dx() + tdir.ortho_dx(reldy);
-    dy = tdir.dy() + tdir.ortho_dy(reldy);
+    coord_translate(face.dir(), reldx, reldy, dx, dy);
 }
 
+// rotates coordinates using two shear operations
+// adjusts angles to work in the range -44.9999 to +44.9999 degrees
+// first shears in the dy direction proportional tan(dir)/2
+// then shears in the dx direction proportional to tan(dir)
+// there is certainly a more mathematically accurate way to do this
+// perhaps something like...
+// https://www.ocf.berkeley.edu/~fricke/projects/israel/paeth/rotation_by_shearing.html
 void vehicle::coord_translate (int dir, int reldx, int reldy, int &dx, int &dy)
 {
-    tileray tdir (dir);
-    tdir.advance (reldx);
-    dx = tdir.dx() + tdir.ortho_dx(reldy);
-    dy = tdir.dy() + tdir.ortho_dy(reldy);
+    int quad = 0;
+    while (dir > 45) {
+        dir = dir - 90;
+        quad++;
+    }
+
+    dir = dir * 0.9999999999; // floating point discontinuity at 45 degrees otherwise
```

Comment:
Good catch on the integer truncation. It still does the job I wanted, just not as cleanly. The goal here was to prevent dir=45 (or 135, etc) from happening. I wanted dir=44.999999, but dir=44 is close enough for now. When I touch this again I'll push a change to do this after converting to a double.